go/ephemeral/lib.go

```diff
@@ -388,6 +388,23 @@ func (e *EKLib) getOrCreateLatestTeamEKInner(ctx context.Context, teamID keybase
 	return teamEK, nil
 }
 
+// Try to get the TeamEK for the given `generation`. If this fails and the
+// `generation` is also the current maxGeneration, create a new teamEK.
+func (e *EKLib) GetMaybeCreateNewTeamEK(ctx context.Context, teamID keybase1.TeamID, generation keybase1.EkGeneration) (teamEK keybase1.TeamEk, err error) {
+	defer e.G().CTrace(ctx, "GetMaybeCreateNewTeamEK", func() error { return err })()
+
+	teamEK, err = e.G().GetTeamEKBoxStorage().Get(ctx, teamID, generation)
+	if err != nil {
+		switch err.(type) {
+		case *EKUnboxErr, *EKMissingBoxErr:
+			if _, cerr := e.GetOrCreateLatestTeamEK(ctx, teamID); cerr != nil {
+				e.G().Log.CDebugf(ctx, "Unable to GetOrCreateLatestTeamEK: %v", cerr)
```

Comment:
The more I think about it, the more it feels a little wrong for the roll to happen here. This only rolls the teamEK when the person who is missing asks for it, and it rolls it in a place where it's not actually about to be used (so the roll might not end up being necessary). It would be better to have the next person who's going to use the latest teamEK, check to see whether it needs to be rolled/reboxed, right before they use it.

Reply:
Let's talk about it after you and Mike are done.